src/parser/json/package_parser.py

```diff
@@ -0,0 +1,203 @@
+                                                                        field=package_dict.get("validUntilDate"),
+                                                                        parsing_method=datetime_from_str,
+                                                                        optional=True)
+
+        version_info: Optional[str] = package_dict.get("versionInfo")
+        raise_parsing_error_if_logger_has_messages(logger, f"Package {name}")
+
+        package = construct_or_raise_parsing_error(Package,
+                                                   dict(spdx_id=spdx_id, name=name, download_location=download_location,
+                                                        version=version_info, file_name=package_file_name,
+                                                        supplier=supplier, originator=originator,
+                                                        files_analyzed=files_analyzed,
+                                                        verification_code=package_verification_code,
+                                                        checksums=checksums, homepage=homepage, source_info=source_info,
+                                                        license_concluded=license_concluded,
+                                                        license_info_from_files=license_info_from_file,
+                                                        license_declared=license_declared,
+                                                        license_comment=license_comments,
+                                                        copyright_text=copyright_text, summary=summary,
+                                                        description=description, comment=comment,
+                                                        external_references=external_refs,
+                                                        attribution_texts=attribution_texts,
+                                                        primary_package_purpose=primary_package_purpose,
+                                                        release_date=release_date, built_date=built_date,
+                                                        valid_until_date=valid_until_date))
+
+        return package
+
+    def parse_external_refs(self, external_ref_dicts: List[Dict]) -> List[ExternalPackageRef]:
+        external_refs = []
+        for external_ref_dict in external_ref_dicts:
+            external_refs = append_parsed_field_or_log_error(logger=self.logger, list_to_append_to=external_refs,
+                                                             field=external_ref_dict, method_to_parse=self.parse_external_ref)
+
+        return external_refs
+
+    def parse_external_ref(self, external_ref_dict: Dict) -> ExternalPackageRef:
+        logger = Logger()
+        ref_category = parse_field_or_log_error(logger=logger, field=external_ref_dict.get("referenceCategory"),
```

Comment:
This will throw an AttributeError if `referenceCategory` is `None`

Reply:
fixed with the changes in `parse_field_or_log_error`, but I also added a testcase for this